### PR TITLE
Cleaning up the TransformComponent

### PR DESCRIPTION
**Pull request by Twolewis**
## What does this PR do?

Removes the giant #if carbonated block, and adds the specific changes to the tx component in.  Better for compatibility changes from o3de

## How was this PR tested?

On PC, editor - making sure the game still runs (since it depends on the netbinding component)

## Notes

Not to be merged in until after sept

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Components/TransformComponent.cpp [hunk 1]
```diff
@@ -382,9 +370,16 @@ namespace AzFramework
 
     void TransformComponent::Deactivate()
     {
-        EBUS_EVENT_ID(m_parentId, AZ::TransformNotificationBus, OnChildRemoved, GetEntityId());
+        AZ::TransformNotificationBus::Event(m_parentId, &AZ::TransformNotificationBus::Events::OnChildRemoved, GetEntityId());
+        auto parentTransform = AZ::TransformBus::FindFirstHandler(m_parentId);
+        if (parentTransform)
+        {
+            parentTransform->NotifyChildChangedEvent(AZ::ChildChangeType::Removed, GetEntityId());
+        }
```
Comment (alexeykostin) [inline Code/Framework/AzFramework/AzFramework/Components/TransformComponent.cpp:378]: These lines did not exist in the original CARBONATED version. Should they be guarded by `#if !defined(CARBONATED)`?

File: Code/Framework/AzCore/AzCore/Component/TransformBus.h [hunk 2]
```diff
@@ -225,20 +225,20 @@ namespace AZ
         virtual AZ::Quaternion GetLocalRotationQuaternion() { return AZ::Quaternion::CreateZero(); }
         //! @}
 
-#if !defined(CARBONATED)
```
Comment (alexeykostin) [inline Code/Framework/AzCore/AzCore/Component/TransformBus.h:228]: `SetLocalScale` and `GetLocalScale` were removed in this PR https://github.com/carbonated-dev/o3de/pull/159. They should not be used, especially `SetLocalScale` that does not exist in O3DE development: https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/TransformBus.h.

Why do you reintroduce them?

File: Code/Framework/AzFramework/AzFramework/Components/TransformComponent.cpp (hunk 1 above)
Comment (Twolewis) [reply Code/Framework/AzFramework/AzFramework/Components/TransformComponent.cpp:378]: If you compare against O3DE 2310.3, they exist there.  It is part of the O3DE engine.
image

File: Code/Framework/AzCore/AzCore/Component/TransformBus.h (hunk 2 above)
Comment (Twolewis) [reply Code/Framework/AzCore/AzCore/Component/TransformBus.h:228]: GetLocalScale exists in O3DE:
https://github.com/o3de/o3de/blob/afd02372f18d2b853756c6c9371da8fb274ab3b6/Code/Framework/AzCore/AzCore/Component/TransformBus.h#L233

You are right about SetLocalScale though - removing

**Review by carbonated-niko — APPROVED** [review]
Thank you, now if I run into weird tx issues, i know its the engine 😆

**Pull request merged**